From 7c710640be1fe59a14d2fd6acfb930ed4cf70995 Mon Sep 17 00:00:00 2001
From: Tomer Eliyahu <tomer.b.eliyahu@intel.com>
Date: Wed, 4 Mar 2020 08:09:33 +0000
Subject: [PATCH] dwpal: fieldValuesGet: remove all ' ' characters

The fieldValuesGet function is used by DWPAL to parse lists of fields of
the format "btm_supported=1", SupportedRates-2 4 11 ", "SSID=", etc.

As part of parsing, the implementation adds ' ' characters to the buffer
for handling special cases, and removes them at the end.

The for loop removing the ' ' characters has a bug since it doesn't take
into account the first ' ', so if an empty value is given such as
"SSID=", the returned value will be ' '.
So, fix it.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>
---
 dwpal.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dwpal.c b/dwpal.c
index d1dcba4..806ad2c 100755
--- a/dwpal.c
+++ b/dwpal.c
@@ -966,7 +966,7 @@ static bool fieldValuesGet(char *buf, size_t bufLen, ParamParsingType parsingTyp
 	}
 
 	/* Remove all ' ' from the end of the string */
-	for (i= idx-1; i > 0; i--)
+	for (i= idx-1; i >= 0; i--)
 	{
 		if (stringOfValues[i] != ' ')
 		{
-- 
2.17.1

